meetings/2021-05-19.md

```diff
@@ -0,0 +1,115 @@
+
+Henry: I'm not arguing soemone would restrict the language. If allowing everyone except X, it's easy to present yourself as anyone but X.
+
+Elf: Some data is not discoverable, so if you would just change identity, you might not get there.
+
+Eric: Even when you have rules of type allow all, you can still flatten them out to conjuctive/disjunctive normal forms. My effective process would be walk through the disjuncts and that allows to see at the end if you have permission. Even with a complex language you can still flatten it to something that has a nice linear performance.
+
+Henry: You could have an ACL access control that links to a document that only gives you positive rules. And others that tells another document will include some negative.
+
+Matthieu: I think one algorhythm would implicitly be able to function in an optimal way without upfront notification of the presence of negative statements.
+
+Henry: If you have linked documents to look through, you can get really long times for resolution.
+
+Justin: I feel like on the other side, a server really can stop as soon as it knows something is not allowed. So the reciprocal stands. It doesn't seem that there is a lot of optimisation or concern on optimisation from the client. We wanna do better than a firewall that just blanks people out. If part of the requirement is that from scratch the client needs to understand everything it can do it's a different kind of system.
+
+Henry: One could have both by having the access control resource specified. I'm ACP or WAC type access resource (I'm regular or geometric logic). And if you state your type of access control, then you have the convenience of knowing upfront the logic order which might be important to notify the user agent about.
+
+Matthieu: It's just a convenient way of expressing certain conditions to add negation and intersection. Say you have access all the time except sunday the 27th from 4-6pm.
+
+Henry: How do we make clear to a client or server guard that they don't have to go through everything?
+
+Justin: We have a use case which essentially gives a useful negation use case (Felicia part of a research group has access to all the resources of the research group but one particular resource she can't because there is a conflict). The thing is if you said group X minus 1 you have to maintain a new group. Now you have two groups to maintain to deal with only one condition. So if you think about other scenarios like this, that can ducktail pretty quickly into many sets being maintained and someone leaves an organisation but you missed a group where they were... This group has access to all RND resources but a couple of people can't and from a manageability point of view, it becomes just more practical. That is where the value of negation really comes through.
```

Comment:
```suggestion
Justin: We have a use case which essentially gives a useful negation use case (Felicia is part of a research group and has access to all the resources of the research group, but one particular resource she can't access because there is a conflict). The thing is if you said group X-minus-1, you have to maintain a new group. Now you have two groups to maintain to deal with only one condition. So if you think about other scenarios like this, that can dovetail pretty quickly into many sets being maintained and someone leaves an organisation but you missed a group where they were a member... This group has access to all RND resources but a couple of people can't and from a manageability point of view, it becomes just more impractical. That is where the value of negation really comes through.
```